dmn-engine/src/main/scala/org/camunda/dmn/evaluation/DecisionTableEvaluator.scala

```diff
@@ -45,43 +48,49 @@ class DecisionTableEvaluator(
 
   def eval(decisionTable: ParsedDecisionTable,
            context: EvalContext): Either[Failure, Val] = {
-    implicit val ctx = context
+    implicit val ctx: EvalContext = context
 
-    evalInputExpressions(decisionTable.inputs).right.flatMap { inputValues =>
-      checkRules(decisionTable.rules, inputValues).right
-        .map(_.flatten)
-        .flatMap { matchedRules =>
-          matchedRules match {
+    class AuditContext(
+        var inputValues: List[Val],
+        var matchedRules: List[(ParsedRule, Iterable[(String, Val)])]
+    )
```

Comment:
To shrink the method a bit, I would suggest moving the class outside of the method and add a `private` modifier.

Reply:
I added the private class to the end of the class - ok?